### PR TITLE
Fix #6136: Don't hardcode default scatter size

### LINKED ISSUE #6136: Feature request: configure default scatter plot marker size
As discussed on the issue tracker, it would be great if there was a way to configure the default size for scatterplot markers.

### PR DESCRIPTION
**Pull request by mdboom**

### CODE REVIEW

File: doc/users/whats_new/style_changes.rst [hunk 1]
```diff
@@ -28,6 +28,13 @@ Colors
 - Grid lines are light grey solid 1pt lines.  They are no longer dashed by
   default.
 
+Plots
+`````
+
+- The default size of the elements in a scatter plot is now based on
+  the rcParam ``lines.markersize`` so it is consistent with ``plot(X,
+  Y, 'o')``.  The old value was 20.
```
Comment (QuLogic) [inline doc/users/whats_new/style_changes.rst]: What's the new value?

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -3726,7 +3726,7 @@ def dopatch(xs, ys, **kwargs):
                                         'facecolors', 'color'],
                          label_namer="y")
     @docstring.dedent_interpd
-    def scatter(self, x, y, s=20, c=None, marker='o', cmap=None, norm=None,
+    def scatter(self, x, y, s=None, c=None, marker='o', cmap=None, norm=None,
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: The documentation for this function also needs updating.

File: doc/users/whats_new/style_changes.rst (hunk 1 above)
Comment (mdboom) [inline doc/users/whats_new/style_changes.rst]: The new value is 36 (6 ^ 2), but it's not hardcoded -- it comes from `lines.markersize`.

**Comment by mdboom** [comment]
Updated for @QuLogic's comments

**Comment by efiring** [comment]
Backported to v2.x as 302c7fd

**Pull request merged**